medicines/web/cypress/integration/tests.js

```diff
@@ -0,0 +1,84 @@
+// We need to replace window.fetch with an XHR-based implementation so that Cypress
+// can mock out the endpoints we use.
+// See https://docs.cypress.io/guides/guides/network-requests.html#Testing-Strategies
+// and https://github.com/cypress-io/cypress/issues/95.
+
+// Some code to do this is adapted from https://github.com/cypress-io/cypress-example-recipes/blob/master/examples/stubbing-spying__window-fetch/cypress/integration/polyfill-fetch-from-tests-spec.js
+
+let polyfill;
+
+// grab fetch polyfill from remote URL, could be also from a local package
+before(() => {
+  const polyfillUrl = 'https://unpkg.com/unfetch/dist/unfetch.umd.js';
+
+  cy.request(polyfillUrl).then(response => {
+    polyfill = response.body;
+  });
+});
+
+Cypress.on('window:before:load', (win) => {
+  delete win.fetch;
+  // since the application code does not ship with a polyfill
+  // load a polyfilled "fetch" from the test
+  win.eval(polyfill);
+  win.fetch = win.unfetch;
+
+  // Clear out session storage so that the disclaimer is always presented.
+  win.sessionStorage.clear();
+})
+
+describe('Search', function() {
+  it('Search for Paracetamol', function() {
+    cy.server();
+    // Mock out first page of search results.
+    cy.route(
+      'https://mhraproductsdev.search.windows.net/indexes/products-index/docs?api-key=CFBCBE8AA11AA871C14001527533870C&api-version=2017-11-11&highlight=content&queryType=full&$count=true&$top=10&$skip=0&search=paracetamol~1+paracetamol^4&scoringProfile=preferKeywords',
+      'fixture:search_results.json',
+    );
+    // Mock out second page of search results.
+    cy.route(
+      'https://mhraproductsdev.search.windows.net/indexes/products-index/docs?api-key=CFBCBE8AA11AA871C14001527533870C&api-version=2017-11-11&highlight=content&queryType=full&$count=true&$top=10&$skip=10&search=paracetamol~1+paracetamol^4&scoringProfile=preferKeywords',
+      'fixture:search_results.json',
+    );
+
+    cy.visit('/');
+    cy.get("input[type='search']").type('paracetamol');
+    cy.contains('Search').click();
+    cy.contains('I have read and understand the disclaimer').click();
+    cy.contains('Agree').click();
+    cy.contains('Next').click();
+    cy.get("a[href='https://example.com/my-cool-document.pdf']");
```

Comment:
This doesn't represent a user action. A user would click on text, and find that it lets them download/view a PDF. They wouldn't look for an anchor tag based on already knowing the `href`.